### PR TITLE
[MISC] Consolidate cleanup() and refactor offline_inference_with_prefix.py

### PR DESCRIPTION
**Pull request by comaniac**
When refactoring offline_inference_with_prefix.py due to #9505, I found that `cleanup()` has been implemented many times in contests. This PR consolidates them to a utility function.

cc @tlrmchlsmth @DarkLight1337 @youkaichao

### CODE REVIEW

File: examples/offline_inference_with_prefix.py [hunk 1]
```diff
@@ -52,6 +48,16 @@
 
 print("-" * 80)
 
+# Destroy the LLM object and free up the GPU memory.
+del regular_llm
+cleanup_dist_env_and_memory()
+
+# The second LLM needs to request a higher gpu_memory_utilization because
+# the first LLM has already allocated a full 30% of the gpu memory.
```
Comment (tlrmchlsmth) [inline examples/offline_inference_with_prefix.py:56]: I think we can delete this now

**Review by tlrmchlsmth — COMMENTED** [review]
should be able to hit `test_lazy_outlines.py` with a cleanup as well:
https://github.com/vllm-project/vllm/blob/ae8b633ba354eaad163e8decf0e4752b5ce58ac2/tests/entrypoints/llm/test_lazy_outlines.py#L17-L34

File: examples/offline_inference_with_prefix.py (hunk 1 above)
Comment (comaniac) [reply examples/offline_inference_with_prefix.py:56]: Ah good catch.

**Review by tlrmchlsmth — APPROVED** [review]
Thanks for the nice cleanup!

**Review by simon-mo — APPROVED** [review]

**Comment by comaniac** [comment]
CI failed due to #9424 which is unrelated

**Pull request merged**